app/com/arpnetworking/metrics/portal/scheduling/JobRef.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright 2018 Dropbox, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.arpnetworking.metrics.portal.scheduling;
+
+import com.arpnetworking.commons.builder.OvalBuilder;
+import models.internal.Organization;
+import models.internal.scheduling.Job;
+import net.sf.oval.constraint.NotNull;
+
+import java.io.Serializable;
+import java.time.Instant;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.function.Supplier;
+
+/**
+ * A serializable reference to a {@link Job}.
+ *
+ * @param <T> The type of the result computed by the job.
+ *
+ * @author Spencer Pearson (spencerpearson at dropbox dot com)
+ */
+public final class JobRef<T> implements Serializable, Supplier<Optional<Job<T>>> {
+    private final JobRepository<T> _repository;
+    private final UUID _id;
+    private final Organization _organization;
+
+    private JobRef(final Builder<T> builder) {
+        _repository = builder._repository;
+        _id = builder._id;
+        _organization = builder._organization;
+    }
+
+    @Override
+    public Optional<Job<T>> get() {
+        return _repository.getJob(_id, _organization);
+    }
+
+    public JobRepository<T> getRepository() {
+        return _repository;
+    }
+
+    public UUID getId() {
+        return _id;
+    }
+
+    public Organization getOrganization() {
+        return _organization;
+    }
+
+    /**
+     * Convenience function to get the last time the job was run from the underlying {@link JobRepository}.
+     *
+     * @return The time.
+     */
+    public Optional<Instant> getLastRun() {
```

Comment:
As a consequence of the indirection described above, these helper methods should probably not exist. However, a method in `JobRef` which accepts an injector and returns the `JobRepository` seems appropriate.